x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingResourceTracker.java

```diff
@@ -242,72 +242,72 @@ static void getMemoryAndProcessors(
             final int numMissingProcessors = numMissingAllocations * numberOfThreadsPerAllocation;
             int numExistingProcessorsToBeUsed = Math.min(numMissingProcessors, numberOfAvailableProcessors);
 
+            if (numberOfRequestedAllocations == 0) {
+                continue;
+            }
             if (assignment.getNodeRoutingTable().isEmpty() == false
                 && assignment.getNodeRoutingTable().values().stream().allMatch(r -> r.getState().consumesMemory() == false)) {
                 // Ignore states that don't consume memory, for example all allocations are failed or stopped
                 // if the node routing table is empty, then it will match the above condition, but it needs to be handled in the next branch
                 continue;
+            }
+
+            if (assignment.getNodeRoutingTable().isEmpty() == false) {
```

Comment:
Everything down here is just indentation.

If you want to review it, I'd recommend settings -> hide whitespace